plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitManager.java

```diff
@@ -367,6 +371,18 @@ public CounterStat getHighMemorySplitSource()
                                 partitionBucketColumns,
                                 partitionBucketCount));
                     }
+                    if (isOptimizeSortedFileReads(session)) {
+                        List<SortingColumn> tableSortedColumns = bucketProperty.get().getSortedBy();
+                        List<SortingColumn> partitionSortedColumns = partitionBucketProperty.get().getSortedBy();
+                        if (!isSortingCompatible(tableSortedColumns, partitionSortedColumns)) {
+                            throw new TrinoException(HIVE_PARTITION_SCHEMA_MISMATCH, format(
+                                    "Hive table (%s) sorting by %s is not compatible with partition (%s) sorting by %s",
```

Comment:
optional: should we add a recommendation for disabling the optimization here in error message?

Reply:
done